### PR TITLE
android: unread message counter

### PR DESCRIPTION
**Pull request by IanRDavies**

### CODE REVIEW

File: apps/android/app/src/main/java/chat/simplex/app/views/helpers/Modifiers.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    val placeable = measurable.measure(constraints)
+
+    // based on the expectation of only one line of text
+    val minPadding = placeable.height / 4
+
+    val width = maxOf(placeable.width + minPadding, placeable.height)
+    layout(width, placeable.height) {
+      placeable.place((width - placeable.width) / 2, 0)
+    }
+  }
```
Comment (IanRDavies) [inline apps/android/app/src/main/java/chat/simplex/app/views/helpers/Modifiers.kt:17]: Copied from https://stackoverflow.com/questions/69987940/jetpack-compose-custom-badge-with-dynamic-text-size-with-circleshape-is-not-draw

File: apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt [hunk 2]
```diff
@@ -23,9 +22,9 @@ import chat.simplex.app.views.chat.item.ChatItemView
 import chat.simplex.app.views.helpers.ChatInfoImage
 import chat.simplex.app.views.helpers.withApi
 import com.google.accompanist.insets.*
-import kotlinx.coroutines.DelicateCoroutinesApi
-import kotlinx.coroutines.launch
+import kotlinx.coroutines.*
```
Comment (epoberezkin) [inline apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt:25]: Please switch to project code styles :)

Comment (epoberezkin) [reply apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt:25]: … and run optimise imports

File: apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt [hunk 3]
```diff
@@ -34,8 +33,25 @@ fun ChatView(chatModel: ChatModel, nav: NavController) {
...
+              CC.ItemRange(chat.chatStats.minUnreadItemId, chat.chatItems.last().id)
+            )
+          }
+        }
+      }
       ChatLayout(chat, chatModel.chatItems,
-        back = { nav.popBackStack() },
+        back = {
+          nav.popBackStack()
+        },
```
Comment (epoberezkin) [inline apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt:54]:
```suggestion
        back = { nav.popBackStack() },
```

File: apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt [hunk 4]
```diff
@@ -84,7 +100,10 @@ fun ChatLayout(
 
 @Composable
 fun ChatInfoToolbar(chat: Chat, back: () -> Unit, info: () -> Unit) {
-  Box(Modifier.height(60.dp).padding(horizontal = 8.dp),
+  Box(
+    Modifier
+      .height(60.dp)
+      .padding(horizontal = 8.dp),
```
Comment (epoberezkin) [inline apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt:106]:
```suggestion
  Box(Modifier.height(60.dp).padding(horizontal = 8.dp),
```

File: apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt [hunk 5]
```diff
@@ -34,8 +33,25 @@ fun ChatView(chatModel: ChatModel, nav: NavController) {
   if (chatModel.chatId.value != null && chatModel.chats.count() > 0) {
     val chat: Chat? = chatModel.chats.firstOrNull { chat -> chat.chatInfo.id == chatModel.chatId.value }
     if (chat != null) {
+
+      // TODO a more advanced version would mark as read only if in view
+      LaunchedEffect(chat.chatItems) {
+        delay(1000L)
```
Comment (epoberezkin) [inline apps/android/app/src/main/java/chat/simplex/app/views/chat/ChatView.kt:39]: here it should check that this is still the current chat

**Review by epoberezkin — APPROVED** [review]

**Review by epoberezkin — APPROVED** [review]

**Pull request merged**